erpnext/manufacturing/doctype/job_card/job_card.py

```diff
@@ -80,6 +80,14 @@ def get_required_items(self):
 					'required_qty': (d.required_qty * flt(self.for_quantity)) / doc.qty
 				})
 
+	def get_scrap_items(self):
+		self.scrap_items = []
+		scrap_items = frappe.get_all('BOM Scrap Item', fields=["item_code", "item_name", "stock_qty", "rate", "amount"],
+			filters={'parenttype': 'BOM', 'parent': self.bom_no}, order_by="idx")
+
+		for item in scrap_items:
+			self.append('scrap_items', item)
```

Comment:
Will this work instead?

```suggestion
		scrap_items = frappe.get_all('BOM Scrap Item',
			filters={'parenttype': 'BOM', 'parent': self.bom_no},
			fields=["item_code", "item_name", "stock_qty", "rate", "amount"],
			order_by="idx")
		self.set('scrap_items', scrap_items)
```